wallet-new/src/Cardano/Wallet/Kernel/Transactions.hs

```diff
@@ -214,7 +229,40 @@ newTransaction ActiveWallet{..} spendingPassword options accountId payees = do
                  Right addr -> pure addr
                  Left err   -> throwM err
 
-
+toMetaForNewTx :: HdAccountId -> TxId -> NonEmpty (TxIn, TxOutAux) -> NonEmpty TxOut -> IO (Maybe TxMeta)
+toMetaForNewTx accountId txId inputs outputs = do
+    case forM inputs toInput of
+        Nothing -> return Nothing
+        Just inputsForMeta -> do
+            time <- Core.getCurrentTimestamp
+            let meta =  TxMeta {
+                          _txMetaId = txId
+                        , _txMetaAmount = minBound -- TODO(kde): find out what this should be. |sum(o) - sum(i)| maybe?
+                        , _txMetaInputs = inputsForMeta
+                        , _txMetaOutputs = aux <$> outputs
+                        , _txMetaCreationAt = time
+                        , _txMetaIsLocal = False -- TODO(kde): find a way to check if all addresses are ours.
```

Comment:
I suggest to take a look at `mkCTx` to figure out what these values should be. We should do whatever the legacy wallet does.